docs/dev-machine-setup.md

```diff
@@ -0,0 +1,109 @@
+
+## Running Firecracker VMM on GCP with nested KVM
+One of the options to set up Firecracker for development purposes is to use a VM on Google Compute Engine (GCE), which supports nested virtualization and allows to run KVM. If you don't have Google Cloud Platform (GCP) account, you can find brief instructions in the Appendix [below](#appendix).
+
+Here is a brief summary of steps to create such a setup, comprising a Ubuntu-based VM on GCE, nested KVM enablement (as documented in GCE [documentation](https://cloud.google.com/compute/docs/instances/enable-nested-virtualization-vm-instances)), as well as Firecracker (following the 'getting started' [instructions](https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md)).
+ 
+  1. Create a new project, so that you can keep track of resources more easily and remove everything easily once your are done. For convenience, give the project a unique name (e.g., <your-username>-firecracker), so that GCP does not need to create project id different than project name (by appending randomized numbers to the name you provide).
+     ```
+     $ gcloud projects create <your-username>-firecracker --enable-cloud-apis --set-as-default
+     ```
+     * Alternatively, if the project you want to use already exists, just switch to the new project using:
+        ```
+        $ gcloud config set project <your-project>
+        ```
+  1. The next step is to create a VM image able to run nested KVM (as outlined [here](https://cloud.google.com/compute/docs/instances/enable-nested-virtualization-vm-instances)). **IMPORTANT:** Notice that Firecracker requires a relatively new kernel, so you should use a recent Ubuntu 18 image (or equivalent).
+     ```
+     $ gcloud compute disks create disk-ub18 --image-project ubuntu-os-cloud --image-family ubuntu-1804-lts
+     ```
+     If you are prompted whether to enable the Compute API for this project, confirm, wait until prompted again, select region - e.g., us-east1-b
+     ```
+     $ gcloud compute images create ub18-nested-kvm --source-disk disk-ub18 --licenses "https://www.googleapis.com/compute/v1/projects/vm-options/global/licenses/enable-vmx" --source-disk-zone us-east1-b
+     ```
+     Make sure you run it as a single command, and that you specify the right region.
+  1. Now we create the VM:
+     ```
+     $ gcloud compute instances create firecracker-vm --zone us-east1-b --image ub18-nested-kvm
+     ```
+  1. Connect to the VM via SSH.  
+     ```
+     $ gcloud compute ssh firecracker-vm
+     ```
+     When doing it for the first time, a key-pair will be created for you (you will be propmpted for a passphrase - can just keep it empty) and uploaded to GCE. Done! You should see the prompt of the new VM: `ubuntu@firecracker-vm:~$`  
+  1. Verify that VMX is enabled, install and enable KVM
+     ```
+     $ grep -cw vmx /proc/cpuinfo
+     1
+     $ sudo apt-get update && sudo apt-get install qemu-kvm -y
```

Comment:
`qemu-kvm` is not needed. Firecracker uses neither qemu, nor any of its userspace tools for kvm.